server/src/main/java/org/apache/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -373,10 +375,18 @@ private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap
     }
   }
 
-  private void loadAllLookupsAndInitStateRef()
+  /**
+   * Load a set of lookups based on the injected value in {@link DataSourceTaskIdHolder#getLookupsToLoad()}.
+   */
+  private void loadLookupsAndInitStateRef()
   {
     List<LookupBean> lookupBeanList = getLookupsList();
     if (lookupBeanList != null) {
+      final List<String> lookupsToLoad = lookupListeningAnnouncerConfig.getLookupsToLoad();
+      final boolean loadAllLookups = lookupsToLoad == null;
+      lookupBeanList = lookupBeanList.stream()
```

Comment:
Nit: We should iterate only if lookupsToLoad is non empty.  Here you are always iterating over the lookup list.

Reply:
Does the below code look fine? If yes, I'll make the change as part of any further commit on this PR.

```java
  List<LookupBean> lookupBeanList = getLookupsList();
    if (lookupBeanList != null) {
      final List<String> lookupsToLoad = lookupListeningAnnouncerConfig.getLookupsToLoad();
      final boolean loadAllLookups = lookupsToLoad == null;
      if (!loadAllLookups){
        lookupBeanList = lookupsToLoad.isEmpty() ? Collections.emptyList() : lookupBeanList.stream()
                                       .filter(lookupBean -> lookupsToLoad.contains(lookupBean.getName()))
                                       .collect(Collectors.toList());
      }
      startLookups(lookupBeanList);
    }
```

Reply:
I am not sure if not doing the iteration is really going to help, performance-wise.

As opposed to this, a more beneficial optimisation would be to not fetch any lookups from coordinator if none need to be loaded.

Reply:
Makes sense, will make this change as part of any further commits on this PR.
Meanwhile, can you please have a look at the other comments on this PR? Thanks!

Reply:
Have made the change.